### PR TITLE
Format units helper proc

### PR DESCRIPTION
**Pull request by Jellyveggie**
Converts the "format_watts" helper proc into a "format_units" helper proc so you can use it on units that aren't watts.
Did CTRL+F to replace all (three) uses of it with the new proc, seems to be working fine. The old "format_watts" is kept around as a wrapper

### CODE REVIEW

File: code/__HELPERS/text.dm [hunk 1]
```diff
@@ -393,6 +393,33 @@ proc/checkhtml(var/t)
 	if(parts.len==2)
 		. += ".[parts[2]]"
 
+
+/**
+ * Formats unites with their suffixes
+ * Should be good for J, W, and stuff
+ */
+var/list/unit_suffixes = list("", "K", "M", "G", "T", "P", "E", "Z", "Y")
```
Comment (PJB3005) [inline code/__HELPERS/text.dm]: Alright time to fix this.

That `K` has to be lower case.

This is very important.

File: code/__HELPERS/text.dm [hunk 2]
```diff
@@ -393,6 +393,33 @@ proc/checkhtml(var/t)
...
+		number/=1000
+		i++
+		if (i == max_unit_suffix)
+			break
+
+	return "[format_num(number)] [unit_suffixes[i]]"
+
+/**
+ * Old unit formatter, the TEG used to use this
+ */
```
Comment (PJB3005) [inline code/__HELPERS/text.dm:422]: Eh you can just remove this one honestly.

**Review by PJB3005 — COMMENTED** [review]
Personally I think it'd be nice if you added a `format_watts` helper which just ran `format_units` and suffixed a W.

**Comment by Jellyveggie** [comment]
augh I had just removed

**Comment by Jellyveggie** [comment]
wait typoed that

File: code/modules/optics/photocollector.dm [hunk 3]
```diff
@@ -57,7 +57,7 @@ var/list/obj/machinery/power/photocollector/photocollector_list = list()
 		return 1
 	else if(istype(W, /obj/item/device/analyzer) || istype(W, /obj/item/device/multitool))
 		if(last_power)
-			to_chat(user, "<span class='notice'>\The [W] registers that [format_watts(last_power)] is being produced every cycle.</span>")
+			to_chat(user, "<span class='notice'>\The [W] registers that [format_units(last_power)]W is being produced every cycle.</span>")
```
Comment (DamianX) [inline code/modules/optics/photocollector.dm]: `format_watts(last_power)` maybe?

File: code/modules/power/generator.dm [hunk 4]
```diff
@@ -347,7 +347,7 @@
 	interface.updateContent("circ1", "Primary circulator ([vertical ? "top"		: "right"])")
 	interface.updateContent("circ2", "Primary circulator ([vertical ? "bottom"	: "left"])")
 
-	interface.updateContent("total_out", format_watts(last_gen))
+	interface.updateContent("total_out", "[format_units(last_gen)]W")
```
Comment (DamianX) [inline code/modules/power/generator.dm]: `format_watts(last_power)` maybe?

File: code/modules/power/singularity/collector.dm [hunk 5]
```diff
@@ -55,7 +55,7 @@ var/global/list/rad_collectors = list()
 		return 1
 	else if(istype(W, /obj/item/device/analyzer) || istype(W, /obj/item/device/multitool))
 		if(active)
-			to_chat(user, "<span class='notice'>\The [W] registers that [format_watts(last_power)] is being produced every cycle.</span>")
+			to_chat(user, "<span class='notice'>\The [W] registers that [format_units(last_power)]W is being produced every cycle.</span>")
```
Comment (DamianX) [inline code/modules/power/singularity/collector.dm]: `format_watts(last_power)` maybe?

**Review by DamianX — COMMENTED** [review]
There's no point in keeping format_watts around if you're not going to use it

**Pull request merged**